polkadot/node/network/approval-distribution/src/lib.rs

```diff
@@ -1475,25 +1670,12 @@ impl State {
 				Some(e) => e,
 			};
 
-			let candidate_entry = match block_entry.candidates.get(index as usize) {
-				None => {
-					gum::debug!(
-					target: LOG_TARGET,
-					?hash,
-					?index,
-					"`get_approval_signatures`: could not find candidate entry for given hash and index!"
-					);
-					continue
-				},
-				Some(e) => e,
-			};
-			let sigs =
-				candidate_entry.messages.iter().filter_map(|(validator_index, message_state)| {
-					match &message_state.approval_state {
-						ApprovalState::Approved(_, sig) => Some((*validator_index, sig.clone())),
-						ApprovalState::Assigned(_) => None,
-					}
-				});
+			let sigs = block_entry
+				.approval_entries(index)
+				.into_iter()
+				.filter_map(|approval_entry| approval_entry.approval(index))
+				.map(|approval| (approval.validator, approval.signature))
+				.collect::<HashMap<ValidatorIndex, ValidatorSignature>>();
```

Comment:
This collect seems superfluous? (Extend below takes any iterator)

Reply:
Will look into optimising it.